gcc/brig/brigfrontend/brig-mem-inst-handler.cc

```diff
@@ -151,14 +155,7 @@ brig_mem_inst_handler::operator () (const BrigBase *base)
 				 address_base, ptr_offset);
 
 	  if (is_store && TREE_TYPE (data) != instr_type)
-	    {
-	      if (int_size_in_bytes (TREE_TYPE (data))
```

Comment:
Again, does this take in account masking away the upper bits? I recall adding this code after nasty bug which resulted from the target optimizer assuming we don't care for the old bits in the upper positions in the target reg.